tests/ui/default_numeric_fallback.stderr

```diff
@@ -0,0 +1,187 @@
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:37:13
+   |
+LL |     let x = 22;
+   |             ^^
+   |
+   = note: `-D clippy::default-numeric-fallback` implied by `-D warnings`
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:38:13
+   |
+LL |     let x = 0.12;
+   |             ^^^^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:39:16
+   |
+LL |     let x: _ = 13;
+   |                ^^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:40:22
+   |
+LL |     let x: [_; 3] = [1, 2, 3];
+   |                      ^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:40:25
+   |
+LL |     let x: [_; 3] = [1, 2, 3];
+   |                         ^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:40:28
+   |
+LL |     let x: [_; 3] = [1, 2, 3];
+   |                            ^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:41:24
+   |
+LL |     let x: (_, i32) = (1, 2);
+   |                        ^
+   |
+   = help: consider adding suffix to avoid default numeric fallback
+
+error: default numeric fallback might occur
+  --> $DIR/default_numeric_fallback.rs:41:27
+   |
+LL |     let x: (_, i32) = (1, 2);
```

Comment:
This is a FP, because the type of the second tuple item has been specified.

Reply:
Let me explain.
Firstly, I thought we had agreed on how this lint should be used. First of all, I thought we both agreed on how the lint was to be used, in that it was to be used in situations where a very high level of reliability was required at the type level, and the appearance of FP was inevitable. Isn't that why we chose the category `restriction'?
The reason this lint is so long is not to detect the possibility of a `Default numeric fallback`, but to try to reduce FP as much as possible while not generating NP.
I suspect your implementation will produce more NP and possibly FP than my implementation.
For example, wouldn't it result in FP in the following case?
```rust
fn foo<T>(t:T) {}
let x: Foo = foo(13);
```